src/libcore/slice/mod.rs

```diff
@@ -1575,6 +1575,134 @@ impl<T> [T] {
         sort::quicksort(self, |a, b| f(a).lt(&f(b)));
     }
 
+    /// Rearranges the elements of the slice such that the following properties hold:
```

Comment:
Since the first line of the docs for an item is duplicated to search results (and to the module page for loose items), it should typically describe the operation succinctly and then elaborate on it in the next lines. You can see this in many of the other methods on `slice` such as the various `sort()` flavors.

As an example (wording could be clearer):

> Reorder the slice so the element at `index` is in its final sorted position.
>
> Partially sorts the slice such that the element at position `index` will be the same as if the array were sorted,
> and any value at position `i < index` will be less than any value at a position `j > index`.

Reply:
Gave it a shot. PTAL.